src/codecs/fifo/fifo.c

```diff
@@ -0,0 +1,46 @@
+#include <stdlib.h>
+
+#include "fifo.h"
+
+typedef struct _fifo_encoder {
+    uint8_t *data;
+    uint16_t len;
+} fifo_encoder;
+
+void fifo_encoder_new(fsk_encoder *encoder)
```

Comment:
This "encoder" could probably be named some other way as it's simply a "container" for data in this case, having nothing specific to any encoding (or use of FIFO in the radio chip).

Maybe it could be just called `raw_encoder` ?

Reply:
Raw encoder makes sense to me